apps/chat/src/utils/app/folders.ts

```diff
@@ -618,3 +619,53 @@ export const getActionsAddFoldersFromFolderId = ({
 
   return actions;
 };
+
+export const isParentFolderSelected = ({
+  currentFolderId,
+  selectedFolderIds,
+}: {
+  currentFolderId: string;
+  selectedFolderIds: string[] | undefined;
+}) => {
+  const parentFolderIds = getParentFolderIdsFromFolderId(currentFolderId);
+
+  const isParentSelected = parentFolderIds.some((id) =>
+    (selectedFolderIds ?? []).includes(`${id}/`),
+  );
+  return isParentSelected;
+};
+
+export const isFolderPartialSelected = ({
+  currentFolderId,
+  partialSelectedFolderIds,
+  isSelected,
+}: {
+  currentFolderId: string;
+  partialSelectedFolderIds: string[] | undefined;
+  isSelected: boolean;
+}) => {
+  const currentId = `${currentFolderId}/`;
+  return !isSelected && (partialSelectedFolderIds ?? []).includes(currentId);
+};
+
+export const getSelectedEntitiesByFolderId = <
+  T extends Conversation | ShareEntity | PublishRequestDialAIEntityModel,
+>({
+  entities,
+  folderId,
+  partialChosenFolderIds,
+  chosenItemsIds,
+}: {
+  entities: T[];
+  folderId: string;
+  partialChosenFolderIds: string[];
+  chosenItemsIds: string[];
+}) =>
+  entities
+    .filter(
+      (entity) =>
+        entity.id.startsWith(folderId) &&
```

Comment:
```suggestion
        entity.id.startsWith(`${folderId}/`) &&
```

should it be like this?